src/components/Combobox.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div class="combobox">
+    <input
+      type="text"
+      :id="id"
+      class="combobox__input"
+      ref="comboboxInput"
+      :value="value"
+      :placeholder="placeholder"
+      @keydown="checkInput"
+      @input="$emit('input', $event.target.value)"
+      autocomplete="off"
+    />
+    <ul v-if="source" class="combobox__options" ref="comboboxList">
+      <li
+        v-for="(option, index) in source"
+        :key="index"
+        class="combobox__option"
+      >
+        {{ option }}
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import Combobo from 'combobo';
+
+export default {
+  name: 'Combobox',
+  props: {
+    id: String,
+    source: Array,
+    value: String,
+    allowCustomInput: {
+      type: Boolean,
+      default: true,
+    },
+    placeholder: String,
+  },
+  methods: {
+    init() {
+      const { comboboxInput, comboboxList } = this.$refs;
+
+      this.combobo = new Combobo({
+        input: comboboxInput,
+        list: comboboxList,
+        options: this.optionEls,
+        activeClass: 'combobox__option--selected',
+      });
+
+      this.combobo.addEventListener('selection', () => {
+        this.$nextTick(() => {
```

Comment:
Why `$nextTick`?

Reply:
For some reason, I was unable to emit the input event on `selection`, waiting for the next tick (DOM update) fixed that.